### PR TITLE
Add dependency on jbwa snapshot, script to release future jbwa snapshots, and method to load the library at runtime

### LINKED ISSUE #1838: Publish jbwa to Broad artifactory
Artifact should contain builds of the library for both Mac and Linux

### PR DESCRIPTION
**Pull request by droazen**
-Published a jbwa snapshot to the Broad artifactory, which we now depend on
 via gradle. This snapshot contains builds of the native jbwa code for both
 Mac and Linux.

-Added utility methods to NativeUtils to load this library at runtime,
 and a test proving that it can be loaded successfully. Also switched
 to the new NativeUtils methods for loading the PairHMM, and confirmed
 that it loads successfully with the HaplotypeCaller in protected

-Included a gradle script to publish a new jbwa snapshot, should it
 become necessary, along with instructions on how to do so.

Resolves #1838

### CODE REVIEW

**Comment by droazen** [comment]
@akiezun or @tomwhite please review

File: build.gradle [hunk 1]
```diff
@@ -148,6 +148,8 @@ dependencies {
         exclude module: 'kryo' // use Spark's version
     }
 
+    compile 'org.broadinstitute:jbwa:bd72b489f9697a857671533c86ad5906967f8672-20160520.183811-1'
```
Comment (droazen) [inline build.gradle]: Note that we are temporarily using an "org.broadinstitute" domain for the jbwa snapshots, but we should switch to a domain controlled by the library author when we start publishing jbwa to maven central.

**Comment by akiezun** [comment]
Here's some code you can add to the test to check that jbwa actually works
```
    @Test
    public void testBwaMemAlignSingleRead() throws Exception {
        final String libraryPath = NativeUtils.runningOnLinux() ? "/lib/libbwajni.so" : "/lib/libbwajni_mac.jnilib";
        Assert.assertTrue(NativeUtils.loadLibraryFromClasspath(libraryPath), "jbwa library was not loaded. " +
                "This could be due to a configuration error, or your system might not support it.");

        BwaIndex index= new BwaIndex(new File(b37_reference_20_21));
        final BwaMem bwaMem = new BwaMem(index);
        //real read taken from src/test/resources/large/CEUTrio.HiSeq.WGS.b37.NA12878.20.21.bam
        final String name="20FUKAAXX100202:3:46:9213:168594";
        final byte[] seqs= "GTTTTGTTTACTACAGCTTTGTAGTAAATTTTGAACTCTAAAGTGTTAGTTCTCTAACTTTGTTTGTTTTTCAAGAGTGTTTTGACTCTTCTTACTGCATC".getBytes(); ;
        final byte[] quals= "DGFDGFDHFFFFGFEFHEGFFFGGHEHFHGHHGEGGGGGFGFHGHGHEHGGGFGAEFGDACAHHDHGCGFGGGFGDHGHFFHDDCGGDGEE".getBytes();
        final ShortRead read = new ShortRead(name, seqs, quals);
        final AlnRgn[] align = bwaMem.align(read);
        Assert.assertEquals(align.length, 1);
        Assert.assertEquals(align[0].getChrom(), "20");
        Assert.assertEquals(align[0].getCigar(), "101M");
        Assert.assertEquals(align[0].getMQual(), 60);
        Assert.assertEquals(align[0].getPos(), 9999997-1); #note difference from the bam file (9999997 in bam, 9999996 here)
        Assert.assertEquals(align[0].getNm(), 0);
        bwaMem.dispose();
    }
```

File: src/test/java/org/broadinstitute/hellbender/JBWAIntegrationTest.java [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+import org.testng.Assert;
+import org.testng.SkipException;
+import org.testng.annotations.Test;
+
+public class JBWAIntegrationTest extends BaseTest {
+
+    @Test
+    public void testJBWAIsLoadable(){
+        if ( ! NativeUtils.runningOnLinux() && ! NativeUtils.runningOnMac() ) {
+            throw new SkipException("jbwa not available on this platform");
```
Comment (akiezun) [inline src/test/java/org/broadinstitute/hellbender/JBWAIntegrationTest.java]: what does this have to do with jbwa? this seems to be a generic check not related to jbwa

**Comment by akiezun** [comment]
looks good - please add 1 test to confirm that jbwa is working and clarify what the first 2 ifs in `testJBWAIsLoadable` have to do with jbwa. BWT I tested on the mac and on linux and it works on both.

File: src/test/java/org/broadinstitute/hellbender/JBWAIntegrationTest.java [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+        }
+
+        if ( NativeUtils.runningOnPPCArchitecture() ) {
+            throw new SkipException("jbwa not available for this architecture");
```
Comment (akiezun) [inline src/test/java/org/broadinstitute/hellbender/JBWAIntegrationTest.java]: this too - does it have anything to do with jbwa itself? if yes, then how?

File: src/test/java/org/broadinstitute/hellbender/JBWAIntegrationTest.java (hunk 2 above)
Comment (akiezun) [inline src/test/java/org/broadinstitute/hellbender/JBWAIntegrationTest.java]: oh, i see. nevermind

**Comment by droazen** [comment]
I've added the additional test you requested, and confirmed that it passes.

The `SkipExceptions` are there to skip JBWA tests on platforms for which we don't have a build of the library -- I've extracted a `skipJBWATestOnUnsupportedPlatforms()` method to make this clearer.

File: build.gradle (hunk 1 above)
Comment (droazen) [inline build.gradle]: Changed to `com.github.lindenb`, and uploaded the same snapshot with the updated groupID.
